Jellyfin.Server/ServerSetupApp/SetupServer.cs

```diff
@@ -0,0 +1,171 @@
+            .ConfigureServices(serv =>
+            {
+                serv.AddHealthChecks()
+                    .AddCheck<SetupHealthcheck>("StartupCheck");
+            })
+            .ConfigureWebHostDefaults(webHostBuilder =>
+                    {
+                        webHostBuilder
+                                .UseKestrel()
+                                .Configure(app =>
+                                {
+                                    app.UseHealthChecks("/health");
+
+                                    app.Map("/startup/logger", loggerRoute =>
+                                    {
+                                        loggerRoute.Run(async context =>
+                                        {
+                                            var networkManager = networkManagerFactory();
+                                            if (context.Connection.RemoteIpAddress is null || networkManager is null || !networkManager.IsInLocalNetwork(context.Connection.RemoteIpAddress))
+                                            {
+                                                context.Response.StatusCode = (int)HttpStatusCode.Unauthorized;
+                                                return;
+                                            }
+
+                                            var logfilePath = Directory.EnumerateFiles(applicationPaths.LogDirectoryPath).Select(e => new FileInfo(e)).OrderBy(f => f.CreationTimeUtc).FirstOrDefault()?.FullName;
+                                            if (logfilePath is not null)
+                                            {
+                                                await context.Response.SendFileAsync(logfilePath, CancellationToken.None).ConfigureAwait(false);
+                                            }
+                                        });
+                                    });
+
+                                    app.Map("/System/Info/Public", systemRoute =>
+                                    {
+                                        systemRoute.Run(async context =>
+                                        {
+                                            var jfApplicationHost = serverApplicationHost();
+
+                                            var retryCounter = 0;
+                                            while (jfApplicationHost is null && retryCounter < 5)
+                                            {
+                                                await Task.Delay(500).ConfigureAwait(false);
+                                                jfApplicationHost = serverApplicationHost();
+                                                retryCounter++;
+                                            }
+
+                                            if (jfApplicationHost is null)
+                                            {
+                                                context.Response.StatusCode = (int)HttpStatusCode.ServiceUnavailable;
```

Comment:
Most SDKs / clients won't parse the response if the status isn't success

Reply:
which is fine? i guess. 
Clients can ignore it for as long as the server is starting, that should all in all result in the same behavior as before for now until they choose to implement that specific waiting behavior.

This is mainly intended for automatic/external systems like watchdogs which will give the _option_ to handle a 503 differently then a 404